File: server/models/officeModel.js
const mongoose = require('mongoose');

const officeSchema = new mongoose.Schema({
    office_address: {
        type: String,
        required: true
    },
    address: {
        type: String,
        required: true
    },
    overview: {
        type: String
    },
    rental_price: {
        type: Number,
        required: true
    },
    wifiAvailable: {
        type: Boolean,
        default: false
    },
    acAvailable: {
        type: Boolean,
        default: false
    },
    cabinsAvailable: {
        type: Number,
        required: true
    },
    max_capacity: {
        type: Number,
        required: true
    },
    contact_number: {
        type: String,
        required: true
    },
    location: {
        type: {
            type: String,
            default: 'Point',
        },
        coordinates: {
            type: [Number],
            required: true
        }
    },
    owner: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    images: {
        type: [String],
    }
});

officeSchema.index({ location: '2dsphere' });

module.exports = mongoose.model('Office', officeSchema);

File: client/lib/models/office_model.dart
class Office {
  final String? id;
  final String officeAddress;
  final String address;
  final String? overview;
  final double rentalPrice;
  final bool wifiAvailable;
  final bool acAvailable;
  final int cabinsAvailable;
  final int maxCapacity;
  final String contactNumber;
  List<double> location;
  final String owner;
  final List<String> images;

  Office({
    this.id,
    required this.officeAddress,
    required this.address,
    this.overview,
    required this.rentalPrice,
    required this.wifiAvailable,
    required this.acAvailable,
    required this.cabinsAvailable,
    required this.maxCapacity,
    required this.contactNumber,
    required this.location,
    required this.owner,
    required this.images,
  });

  factory Office.fromJson(Map<String, dynamic> json) {
    return Office(
      id: json['_id'],
      officeAddress: json['office_address'],
      address: json['address'],
      overview: json['overview'],
      rentalPrice: json['rental_price'].toDouble(),
      wifiAvailable: json['wifiAvailable'],
      acAvailable: json['acAvailable'],
      cabinsAvailable: json['cabinsAvailable'],
      maxCapacity: json['max_capacity'],
      contactNumber: json['contact_number'],
      location: List<double>.from(json['location']['coordinates']),
      owner: json['owner'],
      images: List<String>.from(json['images']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'office_address': officeAddress,
      'address': address,
      'overview': overview,
      'rental_price': rentalPrice,
      'wifiAvailable': wifiAvailable,
      'acAvailable': acAvailable,
      'cabinsAvailable': cabinsAvailable,
      'max_capacity': maxCapacity,
      'contact_number': contactNumber,
      'location': {'coordinates': location},
      'owner': owner,
      'images': images,
    };
  }
}
